### PR TITLE
L112.ghg_en_R_S_T_Y

### PR DESCRIPTION
**Pull request by russellhz**
#340

### CODE REVIEW

File: R/zchunk_L112.ghg_en_R_S_T_Y.R [hunk 1]
```diff
@@ -1,94 +1,173 @@
...
+    EDGAR_sector <- get_data(all_data, "emissions/EDGAR/EDGAR_sector")
     EPA_ghg_tech <- get_data(all_data, "emissions/mappings/EPA_ghg_tech")
-    EDGAR_nation <- get_data(all_data, "emissions/EDGAR_nation")
     GCAM_sector_tech <- get_data(all_data, "emissions/mappings/GCAM_sector_tech")
-    L101.in_EJ_R_en_Si_F_Yh <- get_data(all_data, "L101.in_EJ_R_en_Si_F_Yh")
-    L102.ghg_tgej_USA_en_Sepa_F_2005 <- get_data(all_data, "L102.ghg_tgej_USA_en_Sepa_F_2005")
-    EDGAR_CH4 <- get_data(all_data, "emissions/EDGAR_CH4")
-    EDGAR_N2O <- get_data(all_data, "emissions/EDGAR_N2O")
+    L101.in_EJ_R_en_Si_F_Yh <- get_data(all_data, "L101.in_EJ_R_en_Si_F_Yh") %>%
+      gather(year, energy, `1971`:`2010`)
```
Comment (bpbond) [inline R/zchunk_L112.ghg_en_R_S_T_Y.R]: `gather(year, energy, matches(YEAR_PATTERN))` ?

File: R/zchunk_L112.ghg_en_R_S_T_Y.R [hunk 2]
```diff
@@ -1,94 +1,173 @@
...
+    L102.ghg_tgej_USA_en_Sepa_F_2005 <- get_data(all_data, "L102.ghg_tgej_USA_en_Sepa_F_2005") %>%
+      rename(CH4 = ch4_em_factor, N2O = n2o_em_factor) %>%
+      gather(variable, emiss_factor, CH4, N2O) %>%
+      ungroup
```
Comment (bpbond) [inline R/zchunk_L112.ghg_en_R_S_T_Y.R]: Does this dataset come in grouped? Then let's fix in the upstream chunk.

File: R/zchunk_L112.ghg_en_R_S_T_Y.R [hunk 3]
```diff
@@ -1,94 +1,173 @@
...
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # Computing unscaled EPA emissions by country and technology
+    # This is wrong because by joining only on fuel and sector, not technology, we incorrectly assign
+    # natural gas emissions factors to all passenger transport, even refined liquid passenger transport
```
Comment (bpbond) [inline R/zchunk_L112.ghg_en_R_S_T_Y.R]: Thanks for finding this.

File: R/zchunk_L112.ghg_en_R_S_T_Y.R [hunk 4]
```diff
@@ -1,94 +1,173 @@
...
+                                  select(EPA_agg_sector, EPA_agg_fuel, EDGAR_agg_sector, fuel, sector),
+                                by = c("fuel", "sector")) %>%
+      # Remove years where we don't have emissions data and add Non.CO2 column
+      filter(year %in% emissions.EDGAR_HISTORICAL) %>%
+      repeat_add_columns(tibble(Non.CO2 = c("CH4","N2O"))) %>%
+      # Match in emissions factors
+      # Use left_join because some "cement" only in EPA sector
+      left_join(L102.ghg_tgej_USA_en_Sepa_F_2005,
+                by = c("Non.CO2" = "variable", "EPA_agg_sector" = "sector", "EPA_agg_fuel" = "fuel")) %>%
+      #Compute unscaled emissions
```
Comment (bpbond) [inline R/zchunk_L112.ghg_en_R_S_T_Y.R]: Space after `#`

File: R/zchunk_L201.en_nonco2.R [hunk 5]
```diff
@@ -21,8 +21,8 @@ module_emissions_L201.en_nonco2 <- function(command, ...) {
              FILE = "energy/A_regions",
              FILE = "temp-data-inject/L111.nonghg_tg_R_en_S_F_Yh",
              FILE = "temp-data-inject/L111.nonghg_tgej_R_en_S_F_Yh",
-             FILE = "temp-data-inject/L112.ghg_tg_R_en_S_F_Yh",
```
Comment (bpbond) [inline R/zchunk_L201.en_nonco2.R]: Thanks for doing this!

**Comment by bpbond** [comment]
Hi Marshall, Haewon, can one of you review and/or assign someone else? Thanks.

**Comment by russellhz** [comment]
This is actually an emissions chunk - I forgot to label and add reviewers. I switched Corinne with Marshall

**Comment by bpbond** [comment]
Waiting on review or assignment by @cahartin

**Comment by bpbond** [comment]
Looks like Travis is having some problem at the moment. I'll restart this later.

**Pull request merged**